frontend/src/metabase/collections/components/ModelUploadModal.tsx

```diff
@@ -87,7 +89,7 @@ export function ModelUploadModal({
   useEffect(() => {
     // if we trigger the modal, and there's no uploadable models, just
     // automatically upload a new one
-    if (opened && uploadableModels.length === 0) {
+    if (opened && uploadableModels?.length === 0) {
```

Comment:
This effect triggered too often, even when `uploadableModels` were not loaded yet.
Now there is no default value for it, and we can utilize `undefined` in this condition.